test/SlowTests/Issues/RavenDB-20783.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using FastTests.Server.Replication;
+using Raven.Client.Http;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_20783 : ReplicationTestBase
+    {
+        public RavenDB_20783(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task NullTimerInRecordFastest()
+        {
+            var (servers, leader) = await CreateRaftCluster(3, watcherCluster: true);
+            var mre = new ManualResetEvent(false);
+            var mre2 = new ManualResetEvent(false);
+            using (var store = GetDocumentStore(new Options
+                   {
+                       Server = servers[0],
+                       ModifyDatabaseName = s => $"{s}_1",
+                       ReplicationFactor = 3,
+                       ModifyDocumentStore = s => s.Conventions.ReadBalanceBehavior = Raven.Client.Http.ReadBalanceBehavior.FastestNode
+                   }))
+            {
+                var requestExecutor = store.GetRequestExecutor();
+                NodeSelector node = null;
+                requestExecutor.ForTestingPurposesOnly().OnBeforeScheduleSpeedTest = (n) =>
+                {
+                    node = n;
+                    mre2.Set();
+                    mre.WaitOne();
```

Comment:
Can you do: `Assert.True(mre.WaitOne(some_reasonable_timeout), "nice message")` - we do not want those tests to hang the whole suite

Reply:
done